File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
// - Array.prototype.forEach()

function solution(s){
    var answer = true;
    let open = 0;
    let list = s.split("");
    console.log(list);
    list.forEach(a => {
        if(a === "("){
            open += 1;
        } else if (a === ")"){
            open -= 1;
        }
        if(open < 0){
            answer = false;
        }
    })
    // [실행] 버튼을 누르면 출력 값을 볼 수 있습니다.
    if (open !== 0) {
        return false;
    }
    return answer;
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;

class Solution {
    boolean solution(String s) {
        boolean answer = true;
        
        Stack<Character> st = new Stack<>();
        char[] sArr = s.toCharArray();
        
        for (char a : sArr) {
            if (a == '(') {
                st.push(a);
            } else {
                if (st.empty()) return false;
                st.pop();
            }
        }
        
        if (!st.empty()) return false;

        return answer;
    }
}